### PR TITLE
Partial support of python 3.9

### PR DESCRIPTION
**Pull request by crcrpar**
## Changes

- add Python 3.9 to tests except for integration as there are outstanding libraries for Python 3.9
- add Python 3.9 to Docker build workflow
- enable Docker BuildKit as it better deals `--cache-from`

## TODOs
- monitor integration libraries to keep setup.py and CI workflows up-to-date.

### CODE REVIEW

**Comment by crcrpar** [comment]
I'd prefer squash and merge as I see no benefits to keep all the commit history of this pull request.

**Comment by crcrpar** [comment]
Comment to me: Write comments in example CI job yaml.

File: setup.py
Comment (toshihikoyanase) [inline setup.py:44]: Could you tell me why `Cython` is required for Python 3.9? Some packages are built during installation?

Comment (crcrpar) [reply setup.py:46]: Hope https://github.com/optuna/optuna/pull/1908/commits/481eafaf15b2e680ff0a4090335a59f848c2a26f satisfies you.

Comment (toshihikoyanase) [reply setup.py:46]: Thanks!

**Comment by hvy** [comment]
For the record, some official guidance https://docs.python.org/release/3.9.0/whatsnew/3.9.html#porting-to-python-3-9. Seems like we should be fine without any particular changes to the code.

File: setup.py
Comment (hvy) [reply setup.py:46]: It'd help future maintainers if we know which libraries. I quickly tried this locally on my mac but didn't need Cython so it might be platform specific and coming sometime soon?

**Comment by hvy** [comment]
Changes LGTM at a first glance and after going through the release log. We could also update the top README badge listing the supported Python versions.

**Review by hvy — APPROVED** [review]

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Pull request merged**
